registrations/tests/test_models.py

```diff
@@ -475,6 +476,56 @@ def test_signup_group_registration(self):
             self.contact_person.signup_group.registration_id,
         )
 
+    def test_can_create_access_code(self):
+        self.user = UserFactory(email="test@test.com")
+
+        # Email address is missing.
+        self.assertIsNone(self.contact_person.email)
+        self.assertFalse(self.contact_person.can_create_access_code(self.user))
+
+        # Email address is set and user's email doesn't match it or user doesn't already have
+        # editing and deletion rights for the signup.
+        self.contact_person.email = "test2@test.com"
+        self.contact_person.save(update_fields=["email"])
+        self.assertNotEquals(self.contact_person.email, self.user.email)
```

Comment:
these are quite hard to follow whats going on. Perhaps restructure/split or add  inline comments whats happening 🤔

Reply:
Restructured to individual tests